core/backend/src/BriefcaseManager.ts

```diff
@@ -495,6 +490,14 @@ export class BriefcaseManager {
       imodel.iModelId,
       imodel.changeset?.id,
     );
-    IModelHost.telemetry.postTelemetry(requestContext, telemetryEvent); // eslint-disable-line @typescript-eslint/no-floating-promises
+    activity = activity ?? {
+      activityId: "",
+      applicationId: IModelHost.applicationId,
+      applicationVersion: IModelHost.applicationVersion,
+      sessionId: IModelHost.sessionId,
+      accessToken: "", // IModelHost.getAccessToken(); ACCESS_TOKEN_NEEDS_WORK

```

Comment:
Does this need to be fixed before merging?

Reply:
This has to do with what we report for telemetry from desktop apps about the user identity. Previously it was nothing, and I'm not sure what's right. i don't want to wait for that for this PR

Reply:
The idea that anyone would ever need an access token in a non-async context seems very strange to me.  You should only ever need that token when making a request, which must always be async anyway.

The fact that this was the only such place in this PR seems telling... This function is indeed an edge case, but I would argue that despite appearances, it too is really an async context, it just leaves a dangling promise out there instead of awaiting/returning it.  In fact, I'd almost prefer if this were an async function that callers just knew not to await (it'd certainly make it more testable).  But I can see why we chose not to do that here - it would have required that every call to `logUsage` include that `eslint-disable` comment...

That being said, because this function just leaves a dangling promise, I think it could easily be written to avoid needing to synchronously get a static access token:

```ts
public static logUsage(imodel: IModelDb, activity?: RpcActivity) {
    // eslint-disable-next-line @typescript-eslint/no-floating-promises
    (async () => {
      const telemetryEvent = new TelemetryEvent(
        "imodeljs-backend - Open iModel",
        "7a6424d1-2114-4e89-b13b-43670a38ccd4", // Feature: "iModel Use"
        imodel.iTwinId,
        imodel.iModelId,
        imodel.changeset?.id,
      );
      activity = activity ?? {
        activityId: "",
        applicationId: IModelHost.applicationId,
        applicationVersion: IModelHost.applicationVersion,
        sessionId: IModelHost.sessionId,
        accessToken: await IModelHost.getAccessToken(),
      };

      await IModelHost.telemetry.postTelemetry(activity, telemetryEvent);
    })();
  }
```

Obviously we could refactor that further to get rid of the grossness of the IIFE, but still, I don't think cases like this alone necessitate a way to get access tokens synchronously.

Reply:
@pmconne seems like my comment got lost from the rest of the thread. I normally only do it for top-level async blocks, like in an agent's main.ts file - in other cases, if we have a promise, we probably do care about its result. I don't really see why you'd want to grep something like that, but just for the record, I was only suggesting using it in cases like Bill's snippet above - where you explicitly define that this is a promise you should never await. Maybe I'm naive, but I'd like to believe we're all smart enough to only use `void` when we actually mean it (that we don't want to await the promise) - not when we see an error we don't understand.

Reply:
Recently @kabentley changed a bunch of Tool methods to be async. A ton of code existed that had to ignore floating promises lint rule because methods that should have been async were not. So I had to grep for those and fix them all up.

Basically if we're going to have a lint rule to forbid certain behavior it should be very obvious when we are overriding that rule. `void doSomethingAsync()` is not obvious. Unfortunately it appears the auto-fix for this rule has been changed to prepending `void` to the statement.

Reply:
Applying the auto-fix is actually how I noticed this behavior in the first place. Maybe I do have too much faith in people, I wouldn't have expected them to use `void` in that specific case - it was more like a workaround to put async code into a non-async function than active acknowledgement that you don't, and never will, care about the result or want to wait for it to resolve. But you do have a point, we can't trust everyone to always be cautious about using something like that, so just not using it at all might be a better solution. I'll see if I can find a rule that does this.